wallet/src/ui/app/components/overlay/index.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useCallback } from 'react';
+
+import Icon, { SuiIcons } from '_components/icon';
+
+import type { ReactNode } from 'react';
+
+import st from './Overlay.module.scss';
+
+type OverlayProps = {
+    title: string;
+    className?: string;
+    children: ReactNode | ReactNode[];
+    showModal: boolean;
+    closeOverlay?: () => void;
+    closeIcon?: SuiIcons;
+    setShowModal: (showModal: boolean) => void;
+};
+
+function Overlay({
+    title,
+    className,
+    children,
+    showModal,
+    closeOverlay,
+    setShowModal,
+    closeIcon = SuiIcons.Close,
+}: OverlayProps) {
+    const closeModal = useCallback(
+        (e: React.MouseEvent<HTMLElement>) => {
+            closeOverlay && closeOverlay();
+            setShowModal(false);
+        },
+        [closeOverlay, setShowModal]
+    );
+
+    return (
+        <>
+            {showModal ? (
```

Comment:
When we start updating this to use shared UI components, we should make this use headless UI's dialog for a11y. This is fine for now though.